common/nodes/frontend-server.xml

```diff
@@ -0,0 +1,47 @@
+<stack:stack>
+
+<stack:copyright>
+Copyright (c) 2006 - 2019 Teradata
+All rights reserved. Stacki(r) v5.x stacki.com
+https://github.com/Teradata/stacki/blob/master/LICENSE.txt
+</stack:copyright>
+
+<!-- Make sure to add the frontend box, and enable all added pallets for both the default and frontend box. -->
+<stack:script stack:shell='/opt/stack/bin/python3' stack:stage="install-post" stack:cond="platform != 'aws'">
+import pathlib
+import sys
+import subprocess
+import xml.etree.ElementTree
+import stack.util
+
+# Bail if the roll XML is missing.
+rolls_xml = "/tmp/rolls.xml"
+if not pathlib.Path(rolls_xml).exists():
+	sys.exit(1)
+
+tree = xml.etree.ElementTree.parse(rolls_xml)
+root_element = tree.getroot()
+
+cmd = "/opt/stack/bin/stack"
+# Add pallet can now recursively add everything under a given URL,
+# so only iterate over unique URLs.
+for url in stack.util.unique_everseen(
+	element.attrib["url"] for element in root.iter() if element.attrib.get("url")
```

Comment:
You might just be able to iterate over root directly, instead of calling `iter`. Perhaps this would work?
```
element.attrib["url"] for element in root if "url" in element.attrib
```
It has been a long time since I used ElementTree so you might not be able to treat things like a dict, but worth a shot.

Reply:
I think based on (these docs)[https://docs.python.org/3.8/library/xml.etree.elementtree.html#parsing-xml], specifically the `for child in root:` example, that the `iter()` is needed to recursively iterate through all nodes in the XML.

I'm also doing the `get` because I also want to ignore items where `url` is an empty string.